src/result.sw

```diff
@@ -1,6 +1,59 @@
+//! Error handling with the `Result` type.
+//!
+//! [`Result<T, E>`][`Result`] is the type used for returning and propagating
+//! errors. It is an enum with the variants, [`Ok(T)`], representing
+//! success and containing a value, and [`Err(E)`], representing error
+//! and containing an error value.
+
 library result;
 
+/// `Result` is a type that represents either success ([`Ok`]) or failure
+/// ([`Err`]).
 enum Result<T, E> {
+    /// Contains the success value
     Ok: T,
+
+    /// Contains the error value
     Err: E,
 }
+
+/////////////////////////////////////////////////////////////////////////////
+// Type implementation
+/////////////////////////////////////////////////////////////////////////////
+
+impl Result<T, E> {
+    /////////////////////////////////////////////////////////////////////////
+    // Querying the contained values
+    /////////////////////////////////////////////////////////////////////////
+
+    /// Returns `true` if the result is [`Ok`].
+    fn is_ok(self) -> bool {
+        // TODO the following should compile instead
+        // match self {
+        //     Ok(_) => true,
+        //     _ => false,
+        // }
+        match self {
+            Result::Ok(T) => {
+                true
+            },
+            _ => {
+                false
+            },
+        }
+    }
+
+    /// Returns `true` if the result is [`Err`].
+    fn is_err(self) -> bool {
+        // TODO the following should compile instead
```

Comment:
same here -- i don't think this is important to have here

Reply:
Filed https://github.com/FuelLabs/sway/issues/670